bentoml/_internal/bento/docker/init.sh

```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/env bash
+# Bash Script that installs the dependencies specified in the Bento
+#
+# Usage:
+#   * `init.sh` to run the full script
+#   * `init.sh <step_name>` to run a specific step
+#      available steps: [ensure_python restore_conda_env install_pip_packages install_wheels user_setup_script]
+
+set -ex
+
+# Assert under the root of a Bento directory
+if [ ! -d ./env  ]; then
+    echo "init.sh must be executed from a Bento directory"
+    exit 1
+fi
+
+
+# Check and install the right python version
+if [ $# -eq 0 ] || [ $1 == "ensure_python" ] ; then
+  if [ -f ./python_version ]; then
+    PY_VERSION_SAVED=$(cat ./env/python/version.txt)
+    # remove PATCH version - since most patch version only contains backwards compatible
+    # bug fixes and the BentoML defautl docker base image will include the latest
+    # patch version of each Python minor release
+    DESIRED_PY_VERSION=${PY_VERSION_SAVED:0:3} # returns 3.7, 3.8 or 3.9
+    CURRENT_PY_VERSION=$(python -c 'import sys; print(f"{sys.version_info.major}.{sys.version_info.minor}")')
+
+    if [[ "$DESIRED_PY_VERSION" == "$CURRENT_PY_VERSION" ]]; then
+      echo "Python Version in docker base image $CURRENT_PY_VERSION matches requirement python=$DESIRED_PY_VERSION. Skipping."
+    else
+      if command -v conda >/dev/null 2>&1; then
+        echo "Installing python=$DESIRED_PY_VERSION with conda:"
+        conda install -y -n base pkgs/main::python=$DESIRED_PY_VERSION pip
+      else
+        echo "WARNING: Python Version $DESIRED_PY_VERSION is required, but $CURRENT_PY_VERSION was found."
+      fi
+    fi
+  fi
+fi
+
+if [ $# -eq 0 ] || [ $1 == "restore_conda_env" ] ; then
+  if [ -d ./env/conda ] && [ -f ./env/conda/environment.yml ]; then
+    if command -v conda >/dev/null 2>&1; then
+      echo "Updating conda base environment with environment.yml"
+      # set pip_interop_enabled to improve conda-pip interoperability. Conda can use
+      # pip-installed packages to satisfy dependencies.
+      # this option is only available after conda version 4.6.0
+      # "|| true" ignores the error when the option is not found, for older conda version
+      conda config --set pip_interop_enabled True || true
```

Comment:
re-enabled the pip_interop_enabled option since we migrated away from the other yaml library that was causing issue with it